### PR TITLE
Fix Configuring Learning Rate Schedulers

### LINKED ISSUE #1157: Learning Rate Schedulers' default dictionary parameters should be set via the Trainer
## 🚀 Feature
The default Learning Rate Schedulers (LRS) dictionary parameters should be settable from the Trainer constructor. 

### Motivation
The documentation doesn't seem to be clear that the LRS have the following additional parameters available to be set when you configure the optimizers: 
```
    'interval': 'epoch',  # default every epoch
    'frequency': 1,  # default every epoch/batch
    'reduce_on_plateau': False,  # most often not ReduceLROnPlateau scheduler
    'monitor': 'val_loss'
```
### Pitch
Set those defaults in the constructor of the Trainer and the user can set them themselves. 

### Alternatives
Force the return type of the function LightningModule.configure_optimizers() to be List[Optimizer], List[Dict]

### PR DESCRIPTION
**Pull request by authman**
# Before submitting

- [x] Was this **discussed**/~~approved~~ via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1157 Clarifies within the docs how and where to set scheduler parameters.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
This PR would have saved me an hour, so collectively, I hope it it savs many dev-hours.

At least two others [A](https://github.com/PyTorchLightning/pytorch-lightning/issues/1157), [B](https://github.com/PyTorchLightning/pytorch-lightning/issues/1156) similarly shared confusion about how best and where to set scheduler parameters.

Two places in the docs speak to configuring schedulers:

- [Common Use Cases](https://pytorch-lightning.readthedocs.io/en/0.7.1/optimizers.html#learning-rate-scheduling)
- [Core .configure_optimizers Docs](https://pytorch-lightning.readthedocs.io/en/latest/pytorch_lightning.core.html#pytorch_lightning.core.LightningModule.configure_optimizers)

Neither sufficiently instructs users the appropriate or intended manner of monitoring non-default values or what parameters pytorch searches for to manage schedulers. The functionality exist; it just needs better marketing / documentation =). This PR updates the appropriate .rst files to provide clarity how best to use this functionality.

Docs updated and PEP8 is still in effect.

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -942,28 +942,36 @@ def configure_optimizers(self):
                     dis_sched = CosineAnnealing(discriminator_opt, T_max=10) # called every epoch
                     return [gen_opt, dis_opt], [gen_sched, dis_sched]
 
-        Some things to know
+        Some things to note:
 
-            - Lightning calls ``.backward()`` and ``.step()`` on each optimizer
```
Comment (Borda) [inline pytorch_lightning/core/lightning.py]: why is this listing removed?

Comment (authman) [reply pytorch_lightning/core/lightning.py]: This section used a functionally different formatting in 0.6.0. It rendered properly.



I noticed it was [changed here](https://github.com/PyTorchLightning/pytorch-lightning/commit/f996c2892f60b428b1c5ba4b397fa361103e5c65#diff-2d943a0d0806233840a33725d8a7ce37), which is its current state today =(.

07x

Attempting to understand the intent of the above commit, I believe it was recognized that `.. note::` was being abused and that no other section in the docs looked similar; hence it was removed completely.

I considered the option of bring back `.. note:`, back but under a single consolidated note, which would have rendered properly and is actually my desired outcome, very crisp and clean:

060b

But my commit adds a code block into this note section, and even without it, the note block is already sufficiently large. It really isn't a note but just part of the body/content of the section. I believe the formatting proposed in my current commit works the best.

---

Separately, in the current master / nightly, the entire trainer docs section is missing / isn't built (!) https://pytorch-lightning.readthedocs.io/en/latest/trainer.html due to `WARNING: autodoc: failed to import module ....` while doing a `make html`. However the cause of that is external to this PR.

Comment (Borda) [reply pytorch_lightning/core/lightning.py]: Well to be casted to the same item the new lines has to have one extra space at its beginning...
```
- first line starts 
 and continue here 
```
